packages/hydrooj/src/model/domain.ts

```
@@ -127,14 +128,19 @@ class DomainModel {
     static async countUser(domainId: string, role?: string) {
-        if (role) return await collUser.countDocuments({ domainId, role });
-        return await collUser.countDocuments({ domainId });
+        if (role) return await collUser.countDocuments({ domainId, role, join: true });
+        return await collUser.countDocuments({ domainId, join: true });
     }
 
     @ArgMethod
-    static async setUserRole(domainId: string, uid: MaybeArray<number>, role: string) {
+    static async setUserRole(domainId: string, uid: MaybeArray<number>, role: string, autojoin = false) {
+        const update = { $set: { role, ...(autojoin ? { join: true } : {}) } };
         if (!(uid instanceof Array)) {
```

Biome (1.9.4)

[error] Use Array.isArray() instead of instanceof Array.

instanceof Array returns false for array-like objects and arrays from other execution contexts.
Unsafe fix: Use Array.isArray() instead.


(lint/suspicious/useIsArray)